### PR TITLE
Update podman tun.md `open /dev/net/tun: permission denied`

### PR DESCRIPTION
**Pull request by vic1707**
TL/DR: I found another way to make use of `/dev/net/tun` without running in privileged mode.

---

Hi,
Upon tinkering with my jellyfin container (related to gluetun I promise) I found this [doc](https://jellyfin.org/docs/general/installation/container/#with-hardware-acceleration), while it didn't work in my case due to nvidia I found the idea quite clever and thought a variation of it would work for jellyfin + nvidia on podman.
After a lot of research (I'm missing some of the links to give proper credit, probably lost in the hundreds of browser tabs + history 😓), I fount that `container_use_devices` also exists and might be generic enough to enable my nvidia GPU to work with jellyfin.
It did, yay 🥳.

But after all that I wondered, would that work with gluetun? could i remove the `--privileged` from my compose file if said flag is properly setup ?
It also did !!
my current gluetun compose config
```yml
services:
    gluetun:
        cap_add:
            - NET_ADMIN
        container_name: gluetun
        environment:
            TZ: Europe/Paris
            VPN_SERVICE_PROVIDER: windscribe
            VPN_TYPE: wireguard
        image: docker.io/qmcgaw/gluetun:v3.39.1
        networks:
            shared:
                ipv4_address: 10.99.0.62
        restart: always
        secrets:
            - wireguard_private_key
            - wireguard_addresses
            - wireguard_preshared_key
        security_opt:
            - no-new-privileges=true
        sysctls:
            net.ipv6.conf.all.disable_ipv6: 1
```

---

All this lead me to this PR. I think it needs more testing, more people to try my findings before considering a valid fix/workaround.
there might even be another flag dedicated to net or tun devices, or a `security_opt` label that does it etc... (I'll look into it later, right now my bed is calling me 😴).

Edit: forgot to tell, running on almalinux 9.4 (latest ATM), using podman

### CODE REVIEW

**Comment by mag37** [comment]
Ah that's a good find! I did a similar troubleshooting a while back and ended up with a PR (https://github.com/qdm12/gluetun-wiki/pull/94). 
Your option is much cleaner - I'm not experienced enough to know if either option has any pros or cons.

**Comment by vic1707** [comment]
Hi, 
Didn't see your PR, I'm not experienced enough either, I blindly tried something blindly and it worked 😅.

My solution looks nicer being a one line dummy fix but I feel like yours is more appropriate, I don't really like the idea of changing a global flag (could become a security issue)...

As for your solution, while it's a bit more troublesome I think it's a better one security wise 🤔.

Maybe we can find a good compromise between ease of use and security. Or maybe list both solutions and hope someone with better knowledge can figure it out 🙃.


I'll continue my quest asap. I found out that there isn't a `container_use_tun_devices` or equivalent but maybe some `cap_add` or `security_opt` could do the trick !

**Comment by mag37** [comment]
Yeah listing both solutions might be a good call until someone experienced (or after more testing) can decide what makes either preferred. I just wanted to find a more secure way than using `--privileged` so I'm happy for any solution!

**Comment by vic1707** [comment]
well, just did a podman system reset and retried running gluetun (I have perf issues and try to debug them)
now I get the error of `open /dev/net/tun: permission denied ` guess my fix isn't really one

**Comment by mag37** [comment]
> well, just did a podman system reset and retried running gluetun (I have perf issues and try to debug them) now I get the error of `open /dev/net/tun: permission denied ` guess my fix isn't really one and something happened with podman where running gluetun as privileged caused my server to allow it to then run without it ig 🤷

Well that's odd :/ let me know if you do any more testing and want me to check anything on my side!
